### PR TITLE
Move get_interpolated_setpoint to JointTrajectory from child class

### PR DESCRIPTION
**Pull request by Roelemans**
## Changes

* Move get_interpolated_setpoint to JointTrajectory from child class
* Fix two typos

### CODE REVIEW

File: march_shared_classes/src/march_shared_classes/gait/joint_trajectory.py [hunk 1]
```diff
@@ -68,6 +70,32 @@ def validate_joint_transition(self, joint):
 
         return False
 
+    def get_interpolated_setpoint(self, time):
+        # If we have a setpoint this exact time there is no need to interpolate.
```
Comment (JorisWeeda) [inline march_shared_classes/src/march_shared_classes/gait/joint_trajectory.py]: docstrings? maybe nice to keep consistency

File: march_shared_classes/src/march_shared_classes/gait/joint_trajectory.py [hunk 2]
```diff
@@ -68,6 +70,32 @@ def validate_joint_transition(self, joint):
...
+            if interpolated_setpoints[0][i] > time:
+                position = interpolated_setpoints[1][i - 1]
+                velocity = ((interpolated_setpoints[1][i - 1] - interpolated_setpoints[1][i - 2])
+                            / (interpolated_setpoints[0][i - 1] - interpolated_setpoints[0][i - 2]))
+                return self.setpoint_class(time, position, velocity)
+        rospy.logerr('Could not interpolate setpoint at time {0}'.format(time))
+        return self.setpoint_class(0, 0, 0)
+
+    def interpolate_setpoints(self):
+        time, position, velocity = self.get_setpoints_unzipped()
```
Comment (JorisWeeda) [inline march_shared_classes/src/march_shared_classes/gait/joint_trajectory.py]: docstrings? maybe nice to keep consistency

**Comment by Roelemans** [comment]
I think this PR should be ready for merging now, at least all the basic functionality that worked before works again now. If any new issues pop up, they can be fixed in later PRs. Thanks for the effort boys.

It has been a long time since things were tested this well apparently, because issues have popped up that have been inside develop since at least August. A few issues have already been fixed, others have Jira issues now.

File: march_shared_classes/src/march_shared_classes/gait/joint_trajectory.py (hunk 2 above)
Comment (Roelemans) [reply march_shared_classes/src/march_shared_classes/gait/joint_trajectory.py]: Probably. Something for a later PR unless you want to fix it now

**Comment by Roelemans** [comment]
:tada:

**Pull request merged**